registry/remote/repository.go

```diff
@@ -992,16 +1046,33 @@ func verifyContentDigest(resp *http.Response, expected digest.Digest) error {
 	if digestStr == "" {
 		return nil
 	}
-	contentDigest, err := digest.Parse(digestStr)
-	if err != nil {
+	if contentDigest, err := digest.Parse(digestStr); err != nil {
 		return fmt.Errorf("%s %q: invalid response Docker-Content-Digest: %s", resp.Request.Method, resp.Request.URL, digestStr)
-	}
-	if contentDigest != expected {
+	} else if contentDigest != expected {
 		return fmt.Errorf("%s: mismatch digest: %s", expected, contentDigest)
 	}
 	return nil
 }
 
+// verifyContentDigests is a superset of verifyContentDigest; it takes in the actual (calculated) digest, and
+// verifies against that, the other two digests, both of which may or may not be set to the empty digest.  If empty,
+// there is no validation error, however is not empty, it ensures that the given digest is valid (matches the actual
+// digest)
+func verifyContentDigests(actualDigest digest.Digest, digests map[string]digest.Digest) error {
+	for src, d := range digests {
+		if len(d.String()) == 0 {
```

Comment:
why use an empty string and not `nil`?

Reply:
`d` is a `digest.Digest`, which itself is a `string`.  This is why I have the other PR open in `go-digest`; a `d.IsEmpty()` seems better than `len(d.String()) == 0` or `d.String == ""`.

Reply:
Okay. But why wouldn't we just let it be `nil`?

Reply:
Because you can't set a string to a `nil`; it's not a `*digest.Digest` (pointer). Am I missing something?